tests/deploy/test_scheduler_agent.py

```diff
@@ -1679,3 +1679,34 @@ async def test_scheduler_priority(agent: TestAgent, environment, make_resource_m
     # All tasks were consumed
     queue = agent.scheduler._work.agent_queues._get_queue("agent1")._queue
     assert len(queue) == 0
+
+
+async def test_repair_does_not_trigger_for_blocked_resources(agent: TestAgent, make_resource_minimal):
+    """
+    Ensure that repair doesn't schedule known blocked resources
+    """
+
+    rid1 = ResourceIdStr("test::Resource[agent1,name=1]")
+    rid2 = ResourceIdStr("test::Resource[agent1,name=2]")
+
+    resources = {
+        rid1: make_resource_minimal(rid=rid1, values={"value": "r1_value"}, requires=[]),
+        rid2: make_resource_minimal(rid=rid2, values={"value": "r2_value"}, requires=[]),
+    }
+
+    await agent.scheduler._new_version(version=1, resources=resources, requires=make_requires(resources))
+    await retry_limited(utils.is_agent_done, timeout=5, scheduler=agent.scheduler, agent_name="agent1")
+
+    # Both resources were deployed
+    assert agent.executor_manager.executors["agent1"].execute_count == 2
+
+    agent.executor_manager.reset_executor_counters()
+
+    # mark r1 as blocked and trigger a repair
+    agent.scheduler._state.resource_state[rid1].blocked = BlockedStatus.YES
+
+    await agent.scheduler.repair(reason="Test triggered global repair")
+    await retry_limited(utils.is_agent_done, timeout=5, scheduler=agent.scheduler, agent_name="agent1")
+
+    # Only r2 was deployed
+    assert agent.executor_manager.executors["agent1"].execute_count == 1
```

Comment:
The comment says "only r2" but you only assert r1.

Reply:
Unless I'm missing something, r1 is blocked and only r2 is being redeployed.
Maybe I can improve the wording ? Or change the test to explicitly check that r2 is being deployed (I don't know if there's an easy way to check for this)?

Reply:
My comment was not on point. I had missed that you had both resources on the same agent, so indeed your assertion is correct.